### PR TITLE
fix attach table function name normalizer

### PR DESCRIPTION
**Pull request by sunny19930321**
### Changelog category (leave one): 
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

- Version
   22.3.2.1

### Changelog entry:
- Attach table from local metadata, normalize function name , fix issue: https://github.com/ClickHouse/ClickHouse/pull/36379

### CODE REVIEW

**Comment by vdimir** [comment]
Could you please add some test that shows error?

File: tests/integration/test_attach_table_normalizer/test.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    try:
+        cluster.start()
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+def replace_substring_to_substr(node):
+    node.exec_in_container(["bash", "-c", "sed -i 's/substring/substr/g' /var/lib/clickhouse/metadata/default/file.sql"], user="root")
+
+@pytest.mark.parametrize("engine", ['Ordinary', 'Atomic'])
```
Comment (alesapin) [inline tests/integration/test_attach_table_normalizer/test.py:20]: Why do we need this? Better to add a test with replicated table.

File: src/Interpreters/InterpreterCreateQuery.cpp [hunk 2]
```diff
@@ -953,6 +953,7 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
 
         // Table SQL definition is available even if the table is detached (even permanently)
         auto query = database->getCreateTableQuery(create.getTable(), getContext());
+        FunctionNameNormalizer().visit(query.get());
```
Comment (alesapin) [inline src/Interpreters/InterpreterCreateQuery.cpp:956]: Looks safe, but let's check what we store in ZooKeeper for Replicated tables. Do we normalize values there?

**Review by alesapin — COMMENTED** [review]
Thanks.

File: src/Interpreters/InterpreterCreateQuery.cpp (hunk 2 above)
Comment (sunny19930321) [reply src/Interpreters/InterpreterCreateQuery.cpp:956]: This PR（https://github.com/ClickHouse/ClickHouse/pull/33933） is already compatible with the non-standardized metadata stored by ZK

**Comment by alesapin** [comment]
@mergifyio update

**Comment by sunny19930321** [comment]
> @Mergifyio update

@vdimir there is an upgrade cluster incompatibility issue and this PR is required

**Comment by alexey-milovidov** [comment]
@sunny19930321 the new test appeared to be flaky. Need to fix it.

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 2a126aaa76994ad762589e559ad0dd6dd349beac with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/36384/2a126aaa76994ad762589e559ad0dd6dd349beac/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by alexey-milovidov** [comment]
The tool `black` cannot parse the newly added Python code.

**Comment by sunny19930321** [comment]
> The tool `black` cannot parse the newly added Python code.



> The tool `black` cannot parse the newly added Python code.

Is it the wrong code style？

**Comment by alexey-milovidov** [comment]
Something is wrong:

```
milovidov@milovidov-desktop:~/work/ClickHouse/tests$ git pull --no-rebase git@github.com:sunny19930321/ClickHouse.git fix-attach-table-function-name-normalizer
remote: Enumerating objects: 88, done.
remote: Counting objects: 100% (59/59), done.
remote: Total 88 (delta 58), reused 58 (delta 58), pack-reused 29
Unpacking objects: 100% (88/88), 62.58 KiB | 1.96 MiB/s, done.
From github.com:sunny19930321/ClickHouse
 * branch                    fix-attach-table-function-name-normalizer -> FETCH_HEAD
Auto-merging src/Interpreters/InterpreterCreateQuery.cpp
Merge made by the 'ort' strategy.
 src/Databases/DatabaseOrdinary.cpp                         |  2 ++
 src/Interpreters/InterpreterCreateQuery.cpp                |  1 +
 tests/integration/test_attach_table_normalizer/__init__.py |  0
 tests/integration/test_attach_table_normalizer/test.py     | 59 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 4 files changed, 62 insertions(+)
 create mode 100644 tests/integration/test_attach_table_normalizer/__init__.py
 create mode 100644 tests/integration/test_attach_table_normalizer/test.py
milovidov@milovidov-desktop:~/work/ClickHouse/tests$ cd integration/test_attach_table_normalizer
milovidov@milovidov-desktop:~/work/ClickHouse/tests/integration/test_attach_table_normalizer$ black .
error: cannot format test.py: Cannot parse: 27:12:         user="root"
Oh no! 💥 💔 💥
1 file left unchanged, 1 file failed to reformat.
```

**Comment by alexey-milovidov** [comment]
It is a trivial syntax error.

**Comment by alexey-milovidov** [comment]
Ok, I will merge it in https://github.com/ClickHouse/ClickHouse/pull/51458.

**Pull request closed without merging**